src/es-module-shims.js

```diff
@@ -386,12 +389,31 @@ function processPreload (link) {
   link.ep = true;
   // prepopulate the load record
   const fetchOpts = getFetchOpts(link);
-  // save preloaded fetch options for later load  
+  // save preloaded fetch options for later load
   fetchOptsMap.set(link.href, fetchOpts);
   fetch(link.href, fetchOpts);
 }
 
-function resolve (id, parentUrl) {
+
+async function resolveHook(id, parentUrl) {
+  if (esmsInitOptions.resolve) {
+    const resolved = await esmsInitOptions.resolve(id, parentUrl, defaultResolver);
+
+    if (!resolved) {
+      return defaultResolver(id, parentUrl);
+    }
+
+    if (typeof resolved === 'string') {
+      return { r: resolved }
+    }
+
+    return resolved;
+  }
+
+  return defaultResolver(id, parentUrl);
+}
+
+function defaultResolver (id, parentUrl) {
   const urlResolved = resolveIfNotPlainOrUrl(id, parentUrl);
```

Comment:
This `urlResolved` should be moved into the top of the `resolveHook` as it is supposed to be the comparison of the browser resolver, to determine if import maps are used. Then `defaultResolver` as referenced by the resolve hook will return a string nicely too.